napari/utils/events/event_utils.py

```diff
@@ -23,3 +34,23 @@ def disconnect_events(emitter, listener):
                 listener
             ):
                 em.disconnect(callback)
+
+
+def connect_setattr(emitter: Emitter, obj, attr: str):
+    ref = weakref.ref(obj)
+
+    def _cb(*value):
+        setattr(ref(), attr, value)
+
+    emitter.connect(_cb)
+    # weakref.finalize(obj, emitter.disconnect, _cb)
+
+
+def connect_no_arg(emitter: Emitter, obj, attr: str):
+    ref = weakref.ref(obj)
+
+    def _cb(*_value):
+        getattr(ref(), attr)()
+
+    emitter.connect(_cb)
+    # weakref.finalize(obj, emitter.disconnect, _cb)
```

Comment:
curious, did you run into issues with this?

Reply:
Yes. It produces hundrets of warnings with pyqt5 and casuse segfault with pyside2.